api/config/gateway/v1/gateway.proto

```diff
@@ -16,6 +16,8 @@ message Config {
   repeated Service services = 2;
   repeated Resolver resolvers = 3;
   repeated Module modules = 4;
+
+  string extends = 5;
```

Comment:
i think i would make this repeated so we can aggregate multiple configs easily rather than have to chain configs.